src/components/pages/Offers/Offer/OfferDetails/OfferForm/AccessibilityCheckboxList/AccessibilityCheckboxList.jsx

```diff
@@ -8,46 +8,54 @@ import { ReactComponent as MentalDisabilitySvg } from 'icons/mental-disability.s
 import { ReactComponent as MotorDisabilitySvg } from 'icons/motor-disability.svg'
 import { ReactComponent as VisualDisabilitySvg } from 'icons/visual-disability.svg'
 
-const autoFillValues = function (formValues, field, value) {
-  let disabilityCompliantValues = {
-    noDisabilityCompliant: formValues.noDisabilityCompliant,
-    audioDisabilityCompliant: formValues.audioDisabilityCompliant,
-    mentalDisabilityCompliant: formValues.mentalDisabilityCompliant,
-    motorDisabilityCompliant: formValues.motorDisabilityCompliant,
-    visualDisabilityCompliant: formValues.visualDisabilityCompliant,
+const getDisabilityComplianceValues = values => ({
+  audioDisabilityCompliant: values.audioDisabilityCompliant,
+  mentalDisabilityCompliant: values.mentalDisabilityCompliant,
+  motorDisabilityCompliant: values.motorDisabilityCompliant,
+  visualDisabilityCompliant: values.visualDisabilityCompliant,
+})
+
+const checkHasNoDisabilityCompliance = disabilityComplianceValues => {
+  const hasNull = Object.values(disabilityComplianceValues).includes(null)
+  const hasUndefined = Object.values(disabilityComplianceValues).includes(undefined)
+  if (hasNull || hasUndefined) {
+    return null
   }
 
-  disabilityCompliantValues[field] = value
+  return !Object.values(disabilityComplianceValues).includes(true)
+}
+
+const autoFillValues = function (formValues, field, value) {
+  let noDisabilityCompliant = formValues.noDisabilityCompliant
+  let disabilityCompliantValues = getDisabilityComplianceValues(formValues)
 
   if (field === 'noDisabilityCompliant') {
-    if (disabilityCompliantValues[field]) {
-      disabilityCompliantValues.audioDisabilityCompliant = false
-      disabilityCompliantValues.mentalDisabilityCompliant = false
-      disabilityCompliantValues.motorDisabilityCompliant = false
-      disabilityCompliantValues.visualDisabilityCompliant = false
-    } else {
-      const hasNoDisabilityCompliance = ![
-        disabilityCompliantValues.audioDisabilityCompliant,
-        disabilityCompliantValues.mentalDisabilityCompliant,
-        disabilityCompliantValues.motorDisabilityCompliant,
-        disabilityCompliantValues.visualDisabilityCompliant,
-      ].includes(true)
-      if (hasNoDisabilityCompliance) {
-        disabilityCompliantValues[field] = true
-      }
+    noDisabilityCompliant = value
+
+    if (noDisabilityCompliant) {
+      disabilityCompliantValues = Object.keys(disabilityCompliantValues).reduce(
+        (acc, field) => ({ ...acc, [field]: false }),
+        {}
+      )
+    } else if (!Object.values(disabilityCompliantValues).includes(true)) {
```

Comment:
pour simplifier un peu la lecture dans ce if : 
```js
const hasNoDisabilityComplianceChecked = !Object.values(disabilityCompliantValues).includes(true)
```
```suggestion
    } else if (hasNoDisabilityComplianceChecked) {
```